### PR TITLE
adds decision record for component api style

### PR DESCRIPTION
**Pull request by OskiTheCoder**

### CODE REVIEW

File: documentation/decisions/003_component_api_style.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+status: Accepted
+date: 2023-02-03
+deciders: stephenjwatkins, OskiTheCoder, kevinalexliu
+---
+
+# Design component APIs based on common patterns that exist across popular libraries.
```
Comment (OskiTheCoder) [inline documentation/decisions/003_component_api_style.md:7]: during the spike review, we discussed also having polaris as a reference, should i include that here?

File: documentation/decisions/003_component_api_style.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+1. Picking an existing above library to use as a single point of reference when designing component APIs.
+2. Designing component APIs based on common patterns that exist across popular libraries.
+
+## Decision Outcome
+
+We decided on developing standards and conventions by looking at common API design pattersn that exist across popular component libraries.
+
+## More Information
+
+Many of the component libraries above have some common API design patterns. The advantage of developing conventions based on common patterns is that our component library will be much more intuitive to use since we’d be relying on standards developed by larger teams with stronger support and more resources. In addition, we presently do not have any guidelines for conventions and this decision allows us to initiate efforts immediately with little friction. Conversely, adapting to a single library would require investigation into how the developers of the library approached composition as that influences the overall architecture of the component; and it's not immediately obvious how we could tackle that problem.
```
Comment (OskiTheCoder) [inline documentation/decisions/003_component_api_style.md:44]: after i add the initial component api best practices doc, ill return here to add a reference to that

File: documentation/decisions/003_component_api_style.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- Adobe Spectrum
+
+- GitHub Primer
+
+1. Picking an existing above library to use as a single point of reference when designing component APIs.
+2. Designing component APIs based on common patterns that exist across popular libraries.
+
+## Decision Outcome
+
+We decided on developing standards and conventions by looking at common API design pattersn that exist across popular component libraries.
```
Comment (stephenjwatkins) [inline documentation/decisions/003_component_api_style.md:40]:
```suggestion
We decided on developing standards and conventions by looking at common API design patterns that exist across popular component libraries.
```

File: documentation/decisions/003_component_api_style.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+- Chakra UI
+
+- Ant Design
+
+- Twilio Paste
+
+- Adobe Spectrum
+
+- GitHub Primer
```
Comment (stephenjwatkins) [inline documentation/decisions/003_component_api_style.md:33]: do these need to have an extra space between them or can they be right next to each other?

```suggestion
- MUI
- Shopify Polaris
- Chakra UI
- Ant Design
- Twilio Paste
- Adobe Spectrum
- GitHub Primer
```

File: documentation/decisions/003_component_api_style.md (hunk 1 above)
Comment (stephenjwatkins) [reply documentation/decisions/003_component_api_style.md:7]: i personally don't think we need to mention Polaris by name just yet, even though i agree that they are a reference of inspiration, but i think eventually we can have an Inspiration catalog somewhere in the repo once we have a better feel about that

Comment (stephenjwatkins) [inline documentation/decisions/003_component_api_style.md:7]: curious if you think we can make this heading a little more succinct and without it being in sentence case for consistency with other ADRs?

```suggestion
# Use standard conventions for component APIs
```

File: documentation/decisions/003_component_api_style.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+# Design component APIs based on common patterns that exist across popular libraries.
+
+## Context and Problem Statement
+
+Easy UI should have a set of basic standards and conventions for designing component APIs to deliver consistent DX and UX.
+
+## Decision Drivers
+
+- Scalability: As the project increaes in complexity, conventions should streamline the development process and provide reliability.
```
Comment (stephenjwatkins) [inline documentation/decisions/003_component_api_style.md:15]:
```suggestion
- Scalability: As the project increases in complexity, conventions should streamline the development process and provide reliability.
```

File: documentation/decisions/003_component_api_style.md (hunk 1 above)
Comment (OskiTheCoder) [reply documentation/decisions/003_component_api_style.md:7]: agree

**Review by stephenjwatkins — APPROVED** [review]
looks good!

**Pull request merged**